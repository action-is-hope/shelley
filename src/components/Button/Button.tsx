import React from "react";
/* Adobe libs */
// Version dependancy issue: https://github.com/adobe/react-spectrum/issues/1388#issuecomment-781094658
import { useButton } from "react-aria";
import type { AriaButtonProps } from "@react-types/button";
/* Internal */
import type { Accent, AlignPos, Volume, ButtonVariants } from "../types";
import type { MergeElementProps } from "../utils";
/* Style API */
import { st, classes } from "./button.st.css";

/**
 * Leveraging Adobes 'useButton' to enable use with their hooks.
 * N.B: Omitting 'elementType' as preference is to use existing 'as'.
 *
 * Adobe docs: https://react-spectrum.adobe.com/react-aria/useButton.html
 */

export interface ButtonCustomProps
  extends Omit<AriaButtonProps, "elementType"> {
  /** Define an Icon node, postion via #iconPos. */
  icon?: React.ReactNode;
  /** The position of the icon relative to the label. */
  iconPos?: AlignPos;
  /** Tone index, defines the color palette. */
  tone?: Accent;
  /** Variant index, defines the 'look'. */
  variant?: ButtonVariants;
  /** Defines how 'loud' the Button should be in term of it's size. */
  vol?: Volume;
  /** Applies width of 100%. */
  fullWidth?: boolean;
}

export type ButtonProps<P extends React.ElementType = "button"> = {
  /** Custom element to render such as an anchor "a" or a router "Link" component. */
  as?: P;
  // Dynamically apply element props types based on the input (P).
} & MergeElementProps<P, Omit<ButtonCustomProps, "as">>;

function ButtonBase<T extends React.ElementType = "button">(
  {
    as: As,
    children,
    className: classNameProp,
    icon,
    iconPos = "end",
    fullWidth = false,
    tone = 1,
    variant = "primary",
    vol = 3,
    // Pull off known inputs for @react-aria -> useButton minus 'elementType'
    isDisabled,
    onPress,
    onPressStart,
    onPressEnd,
    onPressChange,
    onPressUp,
    autoFocus,
    onFocus,
    onBlur,
    onFocusChange,
    onKeyDown,
    onKeyUp,
    href,
    target,
    rel,
    ariaExpanded,
    ariaHaspopup,
    ariaControls,
    ariaPressed,
    type,
    id,
    ariaLabel,
    ariaLabelledby,
    ariaDescribedby,
    ariaDetails,
    ...rest
  }: ButtonProps<T>,
  ref: React.Ref<HTMLElement>
) {
  const { buttonProps, isPressed } = useButton(
    {
      isDisabled,
      onPress,
      onPressStart,
      onPressEnd,
      onPressChange,
      onPressUp,
      autoFocus,
      onFocus,
      onBlur,
      onFocusChange,
      onKeyDown,
      onKeyUp,
      href,
      target,
      rel,
      "aria-expanded": ariaExpanded,
      "aria-haspopup": ariaHaspopup,
      "aria-controls": ariaControls,
      "aria-pressed": ariaPressed,
      type,
      id,
      "aria-label": ariaLabel,
      "aria-labelledby": ariaLabelledby,
      "aria-describedby": ariaDescribedby,
      "aria-details": ariaDetails,
      // Map 'as' to elementType for adobe-aria...
      // ...rest,
      elementType: (As as React.JSXElementConstructor<HTMLElement>) || "button",
    },
    ref as React.RefObject<HTMLElement>
  );

  const className = st(
    classes.root,
    {
      iconPos: icon ? iconPos : undefined,
      fullWidth,
      tone,
      variant: variant !== false ? variant : undefined,
      vol: vol !== false ? vol : undefined,
      isPressed,
      isDisabled,
    },
    classNameProp
  );
  const internal = (
    <>
      {icon && (
        <>
          {icon}
          {children && <span className={classes.divider}></span>}
        </>
      )}
      {children && <span className={classes.inner}>{children}</span>}
    </>
  );
  return React.createElement(
    As || "button",
    {
      ref,
      ...buttonProps,
      className,
      ...rest,
    },
    internal
  );
}

// See:
// https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/listbox/src/ListBox.tsx

// const Button = React.forwardRef(ButtonBase);
const Button = React.forwardRef(ButtonBase) as typeof ButtonBase;

// Button.displayName = "Button";

export default Button;
