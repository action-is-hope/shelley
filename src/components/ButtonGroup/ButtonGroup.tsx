import React, {
  Children,
  ReactElement,
  cloneElement,
  forwardRef,
  // Importing from via React. stops EVERYTHING from being dumped out by doc gen.
  // HTMLAttributes,
  isValidElement,
  Ref,
} from "react";
import type { ButtonProps } from "../Button/Button";
import type { Accent, Volume, ButtonVariants } from "../types";
/* = Style API. */
import { st, classes } from "./buttonGroup.st.css";

export interface ButtonGroupProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Adds a class to each button. */
  buttonClassName?: string;
  /** Disables all the buttons. */
  disabled?: boolean;
  /** Tone index. */
  tone?: Accent;
  /** Variant index. */
  variant?: ButtonVariants;
  /** Changes the volume of the buttons. */
  vol?: Volume;
  /** Orient around vertical or horizontal. */
  orientation?: "vertical" | "horizontal";
}

const ButtonGroup = forwardRef(
  (
    {
      buttonClassName,
      children,
      className: classNameProp,
      disabled,
      tone = 1,
      variant = "quiet",
      orientation = "horizontal",
      vol = 3,
      ...rest
    }: ButtonGroupProps,
    ref?: Ref<HTMLDivElement>
  ) => (
    <div
      className={st(classes.root, { orientation }, classNameProp)}
      {...rest}
      ref={ref}
    >
      {Children.map(children, (child) => {
        if (!isValidElement(child)) {
          return null;
        }

        return cloneElement(child as ReactElement, {
          className: st(
            buttonClassName,
            (child.props as ButtonProps)?.className
          ),
          disabled: (child.props as ButtonProps)?.disabled || disabled,
          tone: (child.props as ButtonProps)?.tone || tone,
          vol: (child.props as ButtonProps)?.vol || vol,
          variant: (child.props as ButtonProps)?.variant || variant,
          icon: (child.props as ButtonProps)?.icon,
        });
      })}
    </div>
  )
);

ButtonGroup.displayName = "ButtonGroup";

export default ButtonGroup;
