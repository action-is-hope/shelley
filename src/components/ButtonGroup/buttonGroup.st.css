/* buttonGroup.st.css */

/* = import Button component to style as child. */
@st-import Button from "../Button/button.st.css";

.root {
  -st-states:
    orientation(enum(horizontal, vertical)),
    fullWidth;
  --seperator-color: initial;
  --seperator-width: 1px
}

.root {
  display: inline-flex;
}

.root > Button:focus {
  z-index: 1;
}

/* Styles if `orientation="horizontal"`. */
.root:orientation(horizontal) > Button:not(:first-child),
.root:orientation(horizontal) > Button:not(:first-child)::after {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.root:orientation(horizontal) > Button:not(:last-child),
.root:orientation(horizontal) > Button:not(:last-child)::after {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.root:orientation(horizontal) > Button:not(:first-child) {
  border-left-width: 0;
}

/* Styles if `orientation="vertical"`. */

.root:orientation(vertical) {
  flex-direction: column;
}

.root:orientation(vertical) > Button:not(:first-child),
.root:orientation(vertical) > Button:not(:first-child)::after {
  border-top-right-radius: 0;
  border-top-left-radius: 0;
}

.root:orientation(vertical) > Button:not(:last-child),
.root:orientation(vertical) > Button:not(:last-child)::after {
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
}

.root:orientation(vertical) > Button:not(:first-child) {
  border-top-width: 0;
}

/* = We need to set a border button when in a group. */

/* = horizontal - border-left */
.root:orientation(horizontal) Button:not(:first-child) {
  border-left: var(--seperator-width) solid var(--seperator-color)
}

/* = vertical - border-top */
.root:orientation(vertical) Button:not(:first-child) {
  border-top: var(--seperator-width) solid var(--seperator-color)
}
