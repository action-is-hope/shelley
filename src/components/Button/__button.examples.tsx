/* buttonExamples.tsx */
import React from "react";
/* Example Component */
import Button from "./Button";
/* Supporting Components */
import Icon from "../../components/Icon/Icon";
import PropsDemo from "../../components_site/PropsDemo/PropsDemo";
import CodeSample from "../../components_site/CodeSample/CodeSample";
import { classes as grid } from "../../styles/default/grid.st.css";
export const meta = {
  name: "HTMLInputElement"
};

export const QuickRef = () => (
  <CodeSample
    className={grid.mid}
  >{`import Button from "@action-is-hope/shelley";
\n<Button>Climate fight club</Button>
/* One rule: We don't fight, we act on climate change with intent. */`}</CodeSample>
);

// Props Demo - Examples below.
export const ComponentDemo = () => {
  const [labelDemoProps, setLabelDemoProps]: any = React.useState([
    {
      name: "children",
      label: "children",
      type: "text",
      value: "Earth Solutionists required"
    },
    {
      name: "tone",
      label: "tone",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "variant",
      label: "variant",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "volume",
      label: "volume",
      type: "number",
      min: 0,
      max: 6,
      value: 2
    },
    {
      name: "icon",
      label: "icon",
      type: "select",
      value: "No",
      options: ["Yes", "No"]
    },
    {
      name: "tip",
      label: "tip",
      type: "text",
      value: "@todo - unsure on this..."
    }
  ]);
  return (
    <PropsDemo
      demoProps={labelDemoProps}
      componentName="Button"
      setDemoProps={setLabelDemoProps}
      tsExtends="HTMLButtonElement"
      renderExample={
        <CodeSample>{`import { Button, Icon } from "@action-is-hope/shelley";\n\n<Button ${
          labelDemoProps[1].value > 0
            ? `\n  tone={${labelDemoProps[1].value}}`
            : `\n  tone={false}`
        } ${
          labelDemoProps[2].value > 0
            ? `\n  variant={${labelDemoProps[2].value}}`
            : `\n  variant={false}`
        } ${
          labelDemoProps[3].value > 0
            ? `\n  vol={${labelDemoProps[3].value}}`
            : `\n  vol={false}`
        } ${
          labelDemoProps[4].value === "Yes"
            ? `\n  icon={<Icon><path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path></Icon>}`
            : ``
        }  \n>\n ${labelDemoProps[0].value}\n</Button>`}</CodeSample>
      }
    >
      <Button
        tone={labelDemoProps[1].value === 0 ? false : labelDemoProps[1].value}
        icon={
          labelDemoProps[4].value === "No" ? (
            false
          ) : (
            <Icon alt={labelDemoProps[1].value}>
              {/* crack open an svg to find the path and shove it in. */}
              <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
              {/* <path d="M16 14h-16v-12h16v12zM1 13h14v-10h-14v10z"></path> */}
              {/* <path d="M2 10v2h12v-1c0 0 0.2-1.7-2-2-1.9-0.3-2.2 0.6-3.8 0.6-1.1 0-0.9-1.6-3.2-1.6-1.7 0-3 2-3 2z"></path> */}
              {/* <path d="M13 6c0 1.105-0.895 2-2 2s-2-0.895-2-2c0-1.105 0.895-2 2-2s2 0.895 2 2z"></path> */}
            </Icon>
          )
        }
        variant={
          labelDemoProps[2].value === 0 ? false : labelDemoProps[2].value
        }
        vol={labelDemoProps[3].value === 0 ? false : labelDemoProps[3].value}
        // tip="hi"
      >
        {labelDemoProps[0].value}
      </Button>
      {/* <Button
        tone={labelDemoProps[1].value === 0 ? false : labelDemoProps[1].value}
        icon={
          <Icon>
            <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
          </Icon>
        }
        variant={
          labelDemoProps[2].value === 0 ? false : labelDemoProps[2].value
        }
        vol={labelDemoProps[3].value === 0 ? false : labelDemoProps[3].value}
      >
        &nbsp;
      </Button> */}
    </PropsDemo>
  );
};

export const ComponentHTML = () => (
  <CodeSample>{`<button class="button-root button-tone1 button-variant1 button-vol2">
  <span class="button-inner">Earth Guardians</span>
</button>`}</CodeSample>
);

export const ComponentCSS = () => (
  <CodeSample fixedHeight>{`/** button.st.css */

/* Root */
Button {}

/* Parts */

/* 'inner' button, use for vertical spacing to give a potential Icon more room */
Button::inner {}

/* The divider which is rendered with an icon only. */
Button::divider {}

/* A basic CSS tooltip. */
Button::tip {}


/* Accent colours */
Button.tone1 {}
Button.tone2 {}
Button.tone3 {}
Button.tone4 {}
Button.tone5 {}
Button.tone6 {}

/* Volumes */
Button.vol1 {}
Button.vol2 {}
Button.vol3 {}
Button.vol4 {}
Button.vol5 {}
Button.vol6 {}

/* Variants */
Button.variant1 {}
Button.variant2 {}
Button.variant3 {}
Button.variant4 {}
Button.variant5 {}
Button.variant6 {}`}</CodeSample>
);

// export const Example1 = () => <Demo></Demo>;
// Add some tests to check refs and tings are spread properly.
// const inputEl = React.useRef();
// const test = React.createRef<HTMLButtonElement>();
// const onButtonClick = () => {
//   const node = test.current;
//   // `current` points to the mounted text input element
//   node && node.focus();
// };
