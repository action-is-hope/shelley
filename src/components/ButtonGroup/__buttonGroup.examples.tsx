/* buttonGroupExamples.tsx */
import React from "react";
/* Example Component */
import Button from "../Button/Button";
/* Supporting Components */
import Icon from "../Icon/Icon";
import PropsDemo from "../../components_site/PropsDemo/PropsDemo";
import CodeSample from "../../components_site/CodeSample/CodeSample";
import { classes as grid } from "../../styles/default/grid.st.css";
import ButtonGroup from "./ButtonGroup";
export const meta = {
  name: "ButtonGroup"
};

export const QuickRef = () => (
  <CodeSample
    className={grid.mid}
  >{`import { ButtonGroup, Button } from "@actionishope/shelley";
\n<ButtonGroup vol={2} tone={2} variant={1}>
  <Button>Button one</Button>
  <Button>Button two</Button>
</ButtonGroup>`}</CodeSample>
);

// Props Demo - Examples below.
export const ComponentDemo = () => {
  const [labelDemoProps, setLabelDemoProps]: any = React.useState([
    {
      name: "tone",
      label: "tone",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "variant",
      label: "variant",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "volume",
      label: "volume",
      type: "number",
      min: 0,
      max: 6,
      value: 2
    },
    {
      name: "orientation",
      label: "orientation",
      type: "select",
      value: "horizontal",
      options: ["horizontal", "vertical"]
    }
  ]);
  return (
    <PropsDemo
      demoProps={labelDemoProps}
      componentName={meta.name}
      setDemoProps={setLabelDemoProps}
      tsExtends="HTMLDivElement"
      renderExample={
        <CodeSample>{`import { Button, Icon } from "@actionishope/shelley";\n\n<ButtonGroup ${
          labelDemoProps[0].value > 0
            ? `\n  tone={${labelDemoProps[0].value}}`
            : `\n  tone={false}`
        } ${
          labelDemoProps[1].value > 0
            ? `\n  variant={${labelDemoProps[1].value}}`
            : `\n  variant={false}`
        } ${
          labelDemoProps[2].value > 0
            ? `\n  vol={${labelDemoProps[2].value}}`
            : `\n  vol={false}`
        } ${
          labelDemoProps[3].value === "vertical"
            ? `\n  orientation="vertical"`
            : ``
        } \n>
  <Button>Button one</Button>
  <Button>Button two</Button>
</ButtonGroup>`}</CodeSample>
      }
    >
      <ButtonGroup
        tone={labelDemoProps[0].value === 0 ? false : labelDemoProps[0].value}
        variant={
          labelDemoProps[1].value === 0 ? false : labelDemoProps[1].value
        }
        vol={labelDemoProps[2].value === 0 ? false : labelDemoProps[2].value}
        orientation={labelDemoProps[3].value}
      >
        <Button>Button one</Button>
        <Button>Button two</Button>
      </ButtonGroup>
    </PropsDemo>
  );
};

export const ComponentHTML = () => (
  <CodeSample>{`
<div class="buttonGroup__root buttonGroup---orientation-10-horizontal">
{children}
</div>
`}</CodeSample>
);

export const ComponentCSS = () => (
  <CodeSample fixedHeight>{`/** buttonGroup.st.css */

/* Root */
ButtonGroup {}

ButtonGroup:orientation(vertical) {}

ButtonGroup:orientation(horizontal) {}`}</CodeSample>
);

export const Example1 = () => (
  <>
    <div className={grid.mid}>
      <ButtonGroup vol={2} tone={1} variant={3}>
        <Button>Publish</Button>
        <Button
          icon={
            <Icon alt="Select action">
              {/* < id="chevron-down-small"> */}
              <path d="M8 12l-6.32-6.32 1.67-1.68 4.65 4.65 4.65-4.65 1.67 1.68-6.32 6.32z"></path>
            </Icon>
          }
          aria-label="Page actions"
          aria-haspopup="menu"
        />
      </ButtonGroup>
    </div>

    <CodeSample
      className={grid.mid}
    >{`import { ButtonGroup, Button } from "@actionishope/shelley";
  \n<ButtonGroup vol={2} tone={2} variant={1}>
  <Button>Publish</Button>
  <Button
    icon={
      <Icon alt="Select action">\n<path d="M8 12l-6.32-6.32 1.67-1.68 4.65 4.65 4.65-4.65 1.67 1.68-6.32 6.32z"></path>
      </Icon>
    }
    aria-label="Page actions"
    aria-haspopup="menu"
  />
</ButtonGroup>`}</CodeSample>
  </>
);
