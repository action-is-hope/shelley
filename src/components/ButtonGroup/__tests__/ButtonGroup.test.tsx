// import React from "react";
import Button from "../../Button/Button";
import ButtonGroup from "../ButtonGroup";

import renderer from "react-test-renderer";

describe("ButtonGroup", () => {
  it("renders as a basic primary button group", () => {
    const tree = renderer
      .create(
        <ButtonGroup variant="primary">
          <Button>One</Button>
          <Button>Two</Button>
          <Button>Three</Button>
        </ButtonGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders child buttons with style classes for #tone, #variant, #vol and #buttonClassName", () => {
    const tree = renderer
      .create(
        <ButtonGroup
          variant="secondary"
          tone={6}
          vol={6}
          buttonClassName="button-group-test"
        >
          <Button>One</Button>
          <Button>Two</Button>
          <Button>Three</Button>
        </ButtonGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders vertical button group", () => {
    const tree = renderer
      .create(
        <ButtonGroup variant="quiet" orientation="vertical" vol={1}>
          <Button>One</Button>
          <Button>Two</Button>
          <Button>Three</Button>
        </ButtonGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders button group with all buttons disabled", () => {
    const tree = renderer
      .create(
        <ButtonGroup disabled>
          <Button>One</Button>
          <Button>Two</Button>
          <Button>Three</Button>
        </ButtonGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders buttons with correct style class if overridden", () => {
    const tree = renderer
      .create(
        <ButtonGroup tone={2} vol={4} buttonClassName="btn-group">
          <Button>One</Button>
          <Button tone={3} vol={5} className="btn-group-btn-override">
            Two
          </Button>
          <Button>Three</Button>
        </ButtonGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders buttons with additional class name", () => {
    const tree = renderer
      .create(
        <ButtonGroup vol={4} buttonClassName="btn-group-group-applied">
          <Button>One</Button>
          <Button vol={5} className="btn-group-btn-item-applied">
            Two
          </Button>
          <Button>Three</Button>
        </ButtonGroup>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
