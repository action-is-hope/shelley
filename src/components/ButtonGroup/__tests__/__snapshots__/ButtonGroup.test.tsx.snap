// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup renders as a basic primary button group 1`] = `
<div
  className="buttonGroup85316350__root buttonGroup85316350---orientation-10-horizontal"
>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-7-primary Button368501032---vol-1-3"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      One
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-7-primary Button368501032---vol-1-3"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Two
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-7-primary Button368501032---vol-1-3"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Three
    </span>
  </button>
</div>
`;

exports[`ButtonGroup renders button group with all buttons disabled 1`] = `
<div
  className="buttonGroup85316350__root buttonGroup85316350---orientation-10-horizontal"
>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-3"
    disabled={true}
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      One
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-3"
    disabled={true}
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Two
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-3"
    disabled={true}
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Three
    </span>
  </button>
</div>
`;

exports[`ButtonGroup renders buttons with additional class name 1`] = `
<div
  className="buttonGroup85316350__root buttonGroup85316350---orientation-10-horizontal"
>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-4 btn-group-group-applied"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      One
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-5 btn-group-group-applied btn-group-btn-item-applied"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Two
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-4 btn-group-group-applied"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Three
    </span>
  </button>
</div>
`;

exports[`ButtonGroup renders buttons with correct style class if overridden 1`] = `
<div
  className="buttonGroup85316350__root buttonGroup85316350---orientation-10-horizontal"
>
  <button
    className="Button368501032__root Button368501032---tone-1-2 Button368501032---variant-5-quiet Button368501032---vol-1-4 btn-group"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      One
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-3 Button368501032---variant-5-quiet Button368501032---vol-1-5 btn-group btn-group-btn-override"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Two
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-2 Button368501032---variant-5-quiet Button368501032---vol-1-4 btn-group"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Three
    </span>
  </button>
</div>
`;

exports[`ButtonGroup renders child buttons with style classes for #tone, #variant, #vol and #buttonClassName 1`] = `
<div
  className="buttonGroup85316350__root buttonGroup85316350---orientation-10-horizontal"
>
  <button
    className="Button368501032__root Button368501032---tone-1-6 Button368501032---variant-9-secondary Button368501032---vol-1-6 button-group-test"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      One
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-6 Button368501032---variant-9-secondary Button368501032---vol-1-6 button-group-test"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Two
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-6 Button368501032---variant-9-secondary Button368501032---vol-1-6 button-group-test"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Three
    </span>
  </button>
</div>
`;

exports[`ButtonGroup renders vertical button group 1`] = `
<div
  className="buttonGroup85316350__root buttonGroup85316350---orientation-8-vertical"
>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-1"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      One
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-1"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Two
    </span>
  </button>
  <button
    className="Button368501032__root Button368501032---tone-1-1 Button368501032---variant-5-quiet Button368501032---vol-1-1"
    onBlur={null}
    onClick={[Function]}
    onDragStart={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchCancel={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    type="button"
  >
    <span
      className="Button368501032__inner"
    >
      Three
    </span>
  </button>
</div>
`;
